// @react-aria/calendar/dist/main.js
var $jzHdg$reactarialiveannouncer = require("@react-aria/live-announcer");
var $jzHdg$reactariautils = require("@react-aria/utils");
var $jzHdg$react = require("react");
var $jzHdg$reactariai18n = require("@react-aria/i18n");
var $jzHdg$internationalizeddate = require("@internationalized/date");
var $jzHdg$reactariainteractions = require("@react-aria/interactions");

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "useCalendar", () => $20e845123e697a89$export$3ee915f8151bd4f1);
$parcel$export(module.exports, "useRangeCalendar", () => $c49ada48cbc48220$export$87e0539f600c24e5);
$parcel$export(module.exports, "useCalendarGrid", () => $a07388a797d86b95$export$cb95147730a423f5);
$parcel$export(module.exports, "useCalendarCell", () => $4d833327a32c9193$export$136073280381448e);




var $f87cd00fb2ba0f23$exports = {};
var $1fedb39dedbf7c51$exports = {};
$1fedb39dedbf7c51$exports = JSON.parse("{\"dateRange\":\"{startDate} إلى {endDate}\",\"dateSelected\":\"{date} المحدد\",\"finishRangeSelectionPrompt\":\"انقر لإنهاء عملية تحديد نطاق التاريخ\",\"maximumDate\":\"آخر تاريخ متاح\",\"minimumDate\":\"أول تاريخ متاح\",\"next\":\"التالي\",\"previous\":\"السابق\",\"selectedDateDescription\":\"تاريخ محدد: {date}\",\"selectedRangeDescription\":\"المدى الزمني المحدد: {dateRange}\",\"startRangeSelectionPrompt\":\"انقر لبدء عملية تحديد نطاق التاريخ\",\"todayDate\":\"اليوم، {date}\",\"todayDateSelected\":\"اليوم، {date} محدد\"}");


var $524ada6153e36bf5$exports = {};
$524ada6153e36bf5$exports = JSON.parse("{\"dateRange\":\"{startDate} до {endDate}\",\"dateSelected\":\"Избрано е {date}\",\"finishRangeSelectionPrompt\":\"Натиснете, за да довършите избора на времеви интервал\",\"maximumDate\":\"Последна налична дата\",\"minimumDate\":\"Първа налична дата\",\"next\":\"Напред\",\"previous\":\"Назад\",\"selectedDateDescription\":\"Избрана дата: {date}\",\"selectedRangeDescription\":\"Избран диапазон: {dateRange}\",\"startRangeSelectionPrompt\":\"Натиснете, за да пристъпите към избора на времеви интервал\",\"todayDate\":\"Днес, {date}\",\"todayDateSelected\":\"Днес, {date} са избрани\"}");


var $181053d31fc585a6$exports = {};
$181053d31fc585a6$exports = JSON.parse("{\"dateRange\":\"{startDate} až {endDate}\",\"dateSelected\":\"Vybráno {date}\",\"finishRangeSelectionPrompt\":\"Kliknutím dokončíte výběr rozsahu dat\",\"maximumDate\":\"Poslední dostupné datum\",\"minimumDate\":\"První dostupné datum\",\"next\":\"Další\",\"previous\":\"Předchozí\",\"selectedDateDescription\":\"Vybrané datum: {date}\",\"selectedRangeDescription\":\"Vybrané období: {dateRange}\",\"startRangeSelectionPrompt\":\"Kliknutím zahájíte výběr rozsahu dat\",\"todayDate\":\"Dnes, {date}\",\"todayDateSelected\":\"Dnes, vybráno {date}\"}");


var $87f1f2ee75c25d27$exports = {};
$87f1f2ee75c25d27$exports = JSON.parse("{\"dateRange\":\"{startDate} til {endDate}\",\"dateSelected\":\"{date} valgt\",\"finishRangeSelectionPrompt\":\"Klik for at fuldføre valg af datoområde\",\"maximumDate\":\"Sidste ledige dato\",\"minimumDate\":\"Første ledige dato\",\"next\":\"Næste\",\"previous\":\"Forrige\",\"selectedDateDescription\":\"Valgt dato: {date}\",\"selectedRangeDescription\":\"Valgt interval: {dateRange}\",\"startRangeSelectionPrompt\":\"Klik for at starte valg af datoområde\",\"todayDate\":\"I dag, {date}\",\"todayDateSelected\":\"I dag, {date} valgt\"}");


var $3018d278e45fefcf$exports = {};
$3018d278e45fefcf$exports = JSON.parse("{\"dateRange\":\"{startDate} bis {endDate}\",\"dateSelected\":\"{date} ausgewählt\",\"finishRangeSelectionPrompt\":\"Klicken, um die Auswahl des Datumsbereichs zu beenden\",\"maximumDate\":\"Letztes verfügbares Datum\",\"minimumDate\":\"Erstes verfügbares Datum\",\"next\":\"Weiter\",\"previous\":\"Zurück\",\"selectedDateDescription\":\"Ausgewähltes Datum: {date}\",\"selectedRangeDescription\":\"Ausgewählter Bereich: {dateRange}\",\"startRangeSelectionPrompt\":\"Klicken, um die Auswahl des Datumsbereichs zu beginnen\",\"todayDate\":\"Heute, {date}\",\"todayDateSelected\":\"Heute, {date} ausgewählt\"}");


var $880b624ba6c377b6$exports = {};
$880b624ba6c377b6$exports = JSON.parse("{\"dateRange\":\"{startDate} έως {endDate}\",\"dateSelected\":\"Επιλέχθηκε {date}\",\"finishRangeSelectionPrompt\":\"Κάντε κλικ για να ολοκληρώσετε την επιλογή εύρους ημερομηνιών\",\"maximumDate\":\"Τελευταία διαθέσιμη ημερομηνία\",\"minimumDate\":\"Πρώτη διαθέσιμη ημερομηνία\",\"next\":\"Επόμενο\",\"previous\":\"Προηγούμενο\",\"selectedDateDescription\":\"Επιλεγμένη ημερομηνία: {date}\",\"selectedRangeDescription\":\"Επιλεγμένο εύρος: {dateRange}\",\"startRangeSelectionPrompt\":\"Κάντε κλικ για να ξεκινήσετε την επιλογή εύρους ημερομηνιών\",\"todayDate\":\"Σήμερα, {date}\",\"todayDateSelected\":\"Σήμερα, επιλέχτηκε {date}\"}");


var $6a7e3934431e9523$exports = {};
$6a7e3934431e9523$exports = JSON.parse("{\"previous\":\"Previous\",\"next\":\"Next\",\"selectedDateDescription\":\"Selected Date: {date}\",\"selectedRangeDescription\":\"Selected Range: {dateRange}\",\"todayDate\":\"Today, {date}\",\"todayDateSelected\":\"Today, {date} selected\",\"dateSelected\":\"{date} selected\",\"startRangeSelectionPrompt\":\"Click to start selecting date range\",\"finishRangeSelectionPrompt\":\"Click to finish selecting date range\",\"minimumDate\":\"First available date\",\"maximumDate\":\"Last available date\",\"dateRange\":\"{startDate} to {endDate}\"}");


var $4c458303ff44205c$exports = {};
$4c458303ff44205c$exports = JSON.parse("{\"dateRange\":\"{startDate} a {endDate}\",\"dateSelected\":\"{date} seleccionado\",\"finishRangeSelectionPrompt\":\"Haga clic para terminar de seleccionar rango de fechas\",\"maximumDate\":\"Última fecha disponible\",\"minimumDate\":\"Primera fecha disponible\",\"next\":\"Siguiente\",\"previous\":\"Anterior\",\"selectedDateDescription\":\"Fecha seleccionada: {date}\",\"selectedRangeDescription\":\"Intervalo seleccionado: {dateRange}\",\"startRangeSelectionPrompt\":\"Haga clic para comenzar a seleccionar un rango de fechas\",\"todayDate\":\"Hoy, {date}\",\"todayDateSelected\":\"Hoy, {date} seleccionado\"}");


var $4b6ec4ab19a96fff$exports = {};
$4b6ec4ab19a96fff$exports = JSON.parse("{\"dateRange\":\"{startDate} kuni {endDate}\",\"dateSelected\":\"{date} valitud\",\"finishRangeSelectionPrompt\":\"Klõpsake kuupäevavahemiku valimise lõpetamiseks\",\"maximumDate\":\"Viimane saadaolev kuupäev\",\"minimumDate\":\"Esimene saadaolev kuupäev\",\"next\":\"Järgmine\",\"previous\":\"Eelmine\",\"selectedDateDescription\":\"Valitud kuupäev: {date}\",\"selectedRangeDescription\":\"Valitud vahemik: {dateRange}\",\"startRangeSelectionPrompt\":\"Klõpsake kuupäevavahemiku valimiseks\",\"todayDate\":\"Täna, {date}\",\"todayDateSelected\":\"Täna, {date} valitud\"}");


var $a5184a366ed55e98$exports = {};
$a5184a366ed55e98$exports = JSON.parse("{\"dateRange\":\"{startDate} – {endDate}\",\"dateSelected\":\"{date} valittu\",\"finishRangeSelectionPrompt\":\"Lopeta päivämääräalueen valinta napsauttamalla tätä.\",\"maximumDate\":\"Viimeinen varattavissa oleva päivämäärä\",\"minimumDate\":\"Ensimmäinen varattavissa oleva päivämäärä\",\"next\":\"Seuraava\",\"previous\":\"Edellinen\",\"selectedDateDescription\":\"Valittu päivämäärä: {date}\",\"selectedRangeDescription\":\"Valittu aikaväli: {dateRange}\",\"startRangeSelectionPrompt\":\"Aloita päivämääräalueen valinta napsauttamalla tätä.\",\"todayDate\":\"Tänään, {date}\",\"todayDateSelected\":\"Tänään, {date} valittu\"}");


var $094c11f0389b0f6c$exports = {};
$094c11f0389b0f6c$exports = JSON.parse("{\"dateRange\":\"{startDate} à {endDate}\",\"dateSelected\":\"{date} sélectionné\",\"finishRangeSelectionPrompt\":\"Cliquer pour finir de sélectionner la plage de dates\",\"maximumDate\":\"Dernière date disponible\",\"minimumDate\":\"Première date disponible\",\"next\":\"Suivant\",\"previous\":\"Précédent\",\"selectedDateDescription\":\"Date sélectionnée : {date}\",\"selectedRangeDescription\":\"Plage sélectionnée : {dateRange}\",\"startRangeSelectionPrompt\":\"Cliquer pour commencer à sélectionner la plage de dates\",\"todayDate\":\"Aujourd'hui, {date}\",\"todayDateSelected\":\"Aujourd’hui, {date} sélectionné\"}");


var $5b28c6b8eb99673e$exports = {};
$5b28c6b8eb99673e$exports = JSON.parse("{\"dateRange\":\"{startDate} עד {endDate}\",\"dateSelected\":\"{date} נבחר\",\"finishRangeSelectionPrompt\":\"חץ כדי לסיים את בחירת טווח התאריכים\",\"maximumDate\":\"תאריך פנוי אחרון\",\"minimumDate\":\"תאריך פנוי ראשון\",\"next\":\"הבא\",\"previous\":\"הקודם\",\"selectedDateDescription\":\"תאריך נבחר: {date}\",\"selectedRangeDescription\":\"טווח נבחר: {dateRange}\",\"startRangeSelectionPrompt\":\"לחץ כדי להתחיל בבחירת טווח התאריכים\",\"todayDate\":\"היום, {date}\",\"todayDateSelected\":\"היום, {date} נבחר\"}");


var $519824243c1aad60$exports = {};
$519824243c1aad60$exports = JSON.parse("{\"dateRange\":\"{startDate} do {endDate}\",\"dateSelected\":\"{date} odabran\",\"finishRangeSelectionPrompt\":\"Kliknite da dovršite raspon odabranih datuma\",\"maximumDate\":\"Posljednji raspoloživi datum\",\"minimumDate\":\"Prvi raspoloživi datum\",\"next\":\"Sljedeći\",\"previous\":\"Prethodni\",\"selectedDateDescription\":\"Odabrani datum: {date}\",\"selectedRangeDescription\":\"Odabrani raspon: {dateRange}\",\"startRangeSelectionPrompt\":\"Kliknite da započnete raspon odabranih datuma\",\"todayDate\":\"Danas, {date}\",\"todayDateSelected\":\"Danas, odabran {date}\"}");


var $2fde332abe847288$exports = {};
$2fde332abe847288$exports = JSON.parse("{\"dateRange\":\"{startDate}–{endDate}\",\"dateSelected\":\"{date} kiválasztva\",\"finishRangeSelectionPrompt\":\"Kattintson a dátumtartomány kijelölésének befejezéséhez\",\"maximumDate\":\"Utolsó elérhető dátum\",\"minimumDate\":\"Az első elérhető dátum\",\"next\":\"Következő\",\"previous\":\"Előző\",\"selectedDateDescription\":\"Kijelölt dátum: {date}\",\"selectedRangeDescription\":\"Kijelölt tartomány: {dateRange}\",\"startRangeSelectionPrompt\":\"Kattintson a dátumtartomány kijelölésének indításához\",\"todayDate\":\"Ma, {date}\",\"todayDateSelected\":\"Ma, {date} kijelölve\"}");


var $b266f7fa452e8e83$exports = {};
$b266f7fa452e8e83$exports = JSON.parse("{\"dateRange\":\"Da {startDate} a {endDate}\",\"dateSelected\":\"{date} selezionata\",\"finishRangeSelectionPrompt\":\"Fai clic per completare la selezione dell’intervallo di date\",\"maximumDate\":\"Ultima data disponibile\",\"minimumDate\":\"Prima data disponibile\",\"next\":\"Successivo\",\"previous\":\"Precedente\",\"selectedDateDescription\":\"Data selezionata: {date}\",\"selectedRangeDescription\":\"Intervallo selezionato: {dateRange}\",\"startRangeSelectionPrompt\":\"Fai clic per selezionare l’intervallo di date\",\"todayDate\":\"Oggi, {date}\",\"todayDateSelected\":\"Oggi, {date} selezionata\"}");


var $3528fd03237947e3$exports = {};
$3528fd03237947e3$exports = JSON.parse("{\"dateRange\":\"{startDate} から {endDate}\",\"dateSelected\":\"{date} を選択\",\"finishRangeSelectionPrompt\":\"クリックして日付範囲の選択を終了\",\"maximumDate\":\"最終利用可能日\",\"minimumDate\":\"最初の利用可能日\",\"next\":\"次へ\",\"previous\":\"前へ\",\"selectedDateDescription\":\"選択した日付 : {date}\",\"selectedRangeDescription\":\"選択範囲 : {dateRange}\",\"startRangeSelectionPrompt\":\"クリックして日付範囲の選択を開始\",\"todayDate\":\"本日、{date}\",\"todayDateSelected\":\"本日、{date} を選択\"}");


var $398f3b7902d2708f$exports = {};
$398f3b7902d2708f$exports = JSON.parse("{\"dateRange\":\"{startDate} ~ {endDate}\",\"dateSelected\":\"{date} 선택됨\",\"finishRangeSelectionPrompt\":\"날짜 범위 선택을 완료하려면 클릭하십시오.\",\"maximumDate\":\"마지막으로 사용 가능한 일자\",\"minimumDate\":\"처음으로 사용 가능한 일자\",\"next\":\"다음\",\"previous\":\"이전\",\"selectedDateDescription\":\"선택 일자: {date}\",\"selectedRangeDescription\":\"선택 범위: {dateRange}\",\"startRangeSelectionPrompt\":\"날짜 범위 선택을 시작하려면 클릭하십시오.\",\"todayDate\":\"오늘, {date}\",\"todayDateSelected\":\"오늘, {date} 선택됨\"}");


var $ebaae1c76687311e$exports = {};
$ebaae1c76687311e$exports = JSON.parse("{\"dateRange\":\"Nuo {startDate} iki {endDate}\",\"dateSelected\":\"Pasirinkta {date}\",\"finishRangeSelectionPrompt\":\"Spustelėkite, kad baigtumėte pasirinkti datų intervalą\",\"maximumDate\":\"Paskutinė galima data\",\"minimumDate\":\"Pirmoji galima data\",\"next\":\"Paskesnis\",\"previous\":\"Ankstesnis\",\"selectedDateDescription\":\"Pasirinkta data: {date}\",\"selectedRangeDescription\":\"Pasirinktas intervalas: {dateRange}\",\"startRangeSelectionPrompt\":\"Spustelėkite, kad pradėtumėte pasirinkti datų intervalą\",\"todayDate\":\"Šiandien, {date}\",\"todayDateSelected\":\"Šiandien, pasirinkta {date}\"}");


var $3ecb31d6694563f6$exports = {};
$3ecb31d6694563f6$exports = JSON.parse("{\"dateRange\":\"No {startDate} līdz {endDate}\",\"dateSelected\":\"Atlasīts: {date}\",\"finishRangeSelectionPrompt\":\"Noklikšķiniet, lai pabeigtu datumu diapazona atlasi\",\"maximumDate\":\"Pēdējais pieejamais datums\",\"minimumDate\":\"Pirmais pieejamais datums\",\"next\":\"Tālāk\",\"previous\":\"Atpakaļ\",\"selectedDateDescription\":\"Atlasītais datums: {date}\",\"selectedRangeDescription\":\"Atlasītais diapazons: {dateRange}\",\"startRangeSelectionPrompt\":\"Noklikšķiniet, lai sāktu datumu diapazona atlasi\",\"todayDate\":\"Šodien, {date}\",\"todayDateSelected\":\"Atlasīta šodiena, {date}\"}");


var $26dc2e2c103f178c$exports = {};
$26dc2e2c103f178c$exports = JSON.parse("{\"dateRange\":\"{startDate} til {endDate}\",\"dateSelected\":\"{date} valgt\",\"finishRangeSelectionPrompt\":\"Klikk for å fullføre valg av datoområde\",\"maximumDate\":\"Siste tilgjengelige dato\",\"minimumDate\":\"Første tilgjengelige dato\",\"next\":\"Neste\",\"previous\":\"Forrige\",\"selectedDateDescription\":\"Valgt dato: {date}\",\"selectedRangeDescription\":\"Valgt område: {dateRange}\",\"startRangeSelectionPrompt\":\"Klikk for å starte valg av datoområde\",\"todayDate\":\"I dag, {date}\",\"todayDateSelected\":\"I dag, {date} valgt\"}");


var $6bfd45b83a7d37dd$exports = {};
$6bfd45b83a7d37dd$exports = JSON.parse("{\"dateRange\":\"{startDate} tot {endDate}\",\"dateSelected\":\"{date} geselecteerd\",\"finishRangeSelectionPrompt\":\"Klik om de selectie van het datumbereik te voltooien\",\"maximumDate\":\"Laatste beschikbare datum\",\"minimumDate\":\"Eerste beschikbare datum\",\"next\":\"Volgende\",\"previous\":\"Vorige\",\"selectedDateDescription\":\"Geselecteerde datum: {date}\",\"selectedRangeDescription\":\"Geselecteerd bereik: {dateRange}\",\"startRangeSelectionPrompt\":\"Klik om het datumbereik te selecteren\",\"todayDate\":\"Vandaag, {date}\",\"todayDateSelected\":\"Vandaag, {date} geselecteerd\"}");


var $6f70d036cac4bd46$exports = {};
$6f70d036cac4bd46$exports = JSON.parse("{\"dateRange\":\"{startDate} do {endDate}\",\"dateSelected\":\"Wybrano {date}\",\"finishRangeSelectionPrompt\":\"Kliknij, aby zakończyć wybór zakresu dat\",\"maximumDate\":\"Ostatnia dostępna data\",\"minimumDate\":\"Pierwsza dostępna data\",\"next\":\"Dalej\",\"previous\":\"Wstecz\",\"selectedDateDescription\":\"Wybrana data: {date}\",\"selectedRangeDescription\":\"Wybrany zakres: {dateRange}\",\"startRangeSelectionPrompt\":\"Kliknij, aby rozpocząć wybór zakresu dat\",\"todayDate\":\"Dzisiaj, {date}\",\"todayDateSelected\":\"Dzisiaj wybrano {date}\"}");


var $a227ce57c671783d$exports = {};
$a227ce57c671783d$exports = JSON.parse("{\"dateRange\":\"{startDate} a {endDate}\",\"dateSelected\":\"{date} selecionado\",\"finishRangeSelectionPrompt\":\"Clique para concluir a seleção do intervalo de datas\",\"maximumDate\":\"Última data disponível\",\"minimumDate\":\"Primeira data disponível\",\"next\":\"Próximo\",\"previous\":\"Anterior\",\"selectedDateDescription\":\"Data selecionada: {date}\",\"selectedRangeDescription\":\"Intervalo selecionado: {dateRange}\",\"startRangeSelectionPrompt\":\"Clique para iniciar a seleção do intervalo de datas\",\"todayDate\":\"Hoje, {date}\",\"todayDateSelected\":\"Hoje, {date} selecionado\"}");


var $97acad4758696d87$exports = {};
$97acad4758696d87$exports = JSON.parse("{\"dateRange\":\"{startDate} a {endDate}\",\"dateSelected\":\"{date} selecionado\",\"finishRangeSelectionPrompt\":\"Clique para terminar de selecionar o intervalo de datas\",\"maximumDate\":\"Última data disponível\",\"minimumDate\":\"Primeira data disponível\",\"next\":\"Próximo\",\"previous\":\"Anterior\",\"selectedDateDescription\":\"Data selecionada: {date}\",\"selectedRangeDescription\":\"Intervalo selecionado: {dateRange}\",\"startRangeSelectionPrompt\":\"Clique para começar a selecionar o intervalo de datas\",\"todayDate\":\"Hoje, {date}\",\"todayDateSelected\":\"Hoje, {date} selecionado\"}");


var $010069c7d7fbbea3$exports = {};
$010069c7d7fbbea3$exports = JSON.parse("{\"dateRange\":\"De la {startDate} până la {endDate}\",\"dateSelected\":\"{date} selectată\",\"finishRangeSelectionPrompt\":\"Apăsaţi pentru a finaliza selecţia razei pentru dată\",\"maximumDate\":\"Ultima dată disponibilă\",\"minimumDate\":\"Prima dată disponibilă\",\"next\":\"Următorul\",\"previous\":\"Înainte\",\"selectedDateDescription\":\"Dată selectată: {date}\",\"selectedRangeDescription\":\"Interval selectat: {dateRange}\",\"startRangeSelectionPrompt\":\"Apăsaţi pentru a începe selecţia razei pentru dată\",\"todayDate\":\"Astăzi, {date}\",\"todayDateSelected\":\"Azi, {date} selectată\"}");


var $0ba9c30c48c8d602$exports = {};
$0ba9c30c48c8d602$exports = JSON.parse("{\"dateRange\":\"С {startDate} по {endDate}\",\"dateSelected\":\"Выбрано {date}\",\"finishRangeSelectionPrompt\":\"Щелкните, чтобы завершить выбор диапазона дат\",\"maximumDate\":\"Последняя доступная дата\",\"minimumDate\":\"Первая доступная дата\",\"next\":\"Далее\",\"previous\":\"Назад\",\"selectedDateDescription\":\"Выбранная дата: {date}\",\"selectedRangeDescription\":\"Выбранный диапазон: {dateRange}\",\"startRangeSelectionPrompt\":\"Щелкните, чтобы начать выбор диапазона дат\",\"todayDate\":\"Сегодня, {date}\",\"todayDateSelected\":\"Сегодня, выбрано {date}\"}");


var $6873e95bd7af7559$exports = {};
$6873e95bd7af7559$exports = JSON.parse("{\"dateRange\":\"Od {startDate} do {endDate}\",\"dateSelected\":\"Vybratý dátum {date}\",\"finishRangeSelectionPrompt\":\"Kliknutím dokončíte výber rozsahu dátumov\",\"maximumDate\":\"Posledný dostupný dátum\",\"minimumDate\":\"Prvý dostupný dátum\",\"next\":\"Nasledujúce\",\"previous\":\"Predchádzajúce\",\"selectedDateDescription\":\"Vybratý dátum: {date}\",\"selectedRangeDescription\":\"Vybratý rozsah: {dateRange}\",\"startRangeSelectionPrompt\":\"Kliknutím spustíte výber rozsahu dátumov\",\"todayDate\":\"Dnes {date}\",\"todayDateSelected\":\"Vybratý dnešný dátum {date}\"}");


var $ce3c7852583668e3$exports = {};
$ce3c7852583668e3$exports = JSON.parse("{\"dateRange\":\"{startDate} do {endDate}\",\"dateSelected\":\"{date} izbrano\",\"finishRangeSelectionPrompt\":\"Kliknite za dokončanje izbire datumskega obsega\",\"maximumDate\":\"Zadnji razpoložljivi datum\",\"minimumDate\":\"Prvi razpoložljivi datum\",\"next\":\"Naprej\",\"previous\":\"Nazaj\",\"selectedDateDescription\":\"Izbrani datum: {date}\",\"selectedRangeDescription\":\"Izbrano območje: {dateRange}\",\"startRangeSelectionPrompt\":\"Kliknite za začetek izbire datumskega obsega\",\"todayDate\":\"Danes, {date}\",\"todayDateSelected\":\"Danes, {date} izbrano\"}");


var $36cbdcce92af5213$exports = {};
$36cbdcce92af5213$exports = JSON.parse("{\"dateRange\":\"{startDate} do {endDate}\",\"dateSelected\":\"{date} izabran\",\"finishRangeSelectionPrompt\":\"Kliknite da dovršite opseg izabranih datuma\",\"maximumDate\":\"Zadnji raspoloživi datum\",\"minimumDate\":\"Prvi raspoloživi datum\",\"next\":\"Sledeći\",\"previous\":\"Prethodni\",\"selectedDateDescription\":\"Izabrani datum: {date}\",\"selectedRangeDescription\":\"Izabrani period: {dateRange}\",\"startRangeSelectionPrompt\":\"Kliknite da započnete opseg izabranih datuma\",\"todayDate\":\"Danas, {date}\",\"todayDateSelected\":\"Danas, izabran {date}\"}");


var $ce53fa032c8abdc9$exports = {};
$ce53fa032c8abdc9$exports = JSON.parse("{\"dateRange\":\"{startDate} till {endDate}\",\"dateSelected\":\"{date} har valts\",\"finishRangeSelectionPrompt\":\"Klicka för att avsluta val av datumintervall\",\"maximumDate\":\"Sista tillgängliga datum\",\"minimumDate\":\"Första tillgängliga datum\",\"next\":\"Nästa\",\"previous\":\"Föregående\",\"selectedDateDescription\":\"Valt datum: {date}\",\"selectedRangeDescription\":\"Valt intervall: {dateRange}\",\"startRangeSelectionPrompt\":\"Klicka för att välja datumintervall\",\"todayDate\":\"Idag, {date}\",\"todayDateSelected\":\"Idag, {date} har valts\"}");


var $358703e281d51beb$exports = {};
$358703e281d51beb$exports = JSON.parse("{\"dateRange\":\"{startDate} - {endDate}\",\"dateSelected\":\"{date} seçildi\",\"finishRangeSelectionPrompt\":\"Tarih aralığı seçimini tamamlamak için tıklayın\",\"maximumDate\":\"Son müsait tarih\",\"minimumDate\":\"İlk müsait tarih\",\"next\":\"Sonraki\",\"previous\":\"Önceki\",\"selectedDateDescription\":\"Seçilen Tarih: {date}\",\"selectedRangeDescription\":\"Seçilen Aralık: {dateRange}\",\"startRangeSelectionPrompt\":\"Tarih aralığı seçimini başlatmak için tıklayın\",\"todayDate\":\"Bugün, {date}\",\"todayDateSelected\":\"Bugün, {date} seçildi\"}");


var $16bef71241a84fd2$exports = {};
$16bef71241a84fd2$exports = JSON.parse("{\"dateRange\":\"{startDate} — {endDate}\",\"dateSelected\":\"Вибрано {date}\",\"finishRangeSelectionPrompt\":\"Натисніть, щоб завершити вибір діапазону дат\",\"maximumDate\":\"Остання доступна дата\",\"minimumDate\":\"Перша доступна дата\",\"next\":\"Наступний\",\"previous\":\"Попередній\",\"selectedDateDescription\":\"Вибрана дата: {date}\",\"selectedRangeDescription\":\"Вибраний діапазон: {dateRange}\",\"startRangeSelectionPrompt\":\"Натисніть, щоб почати вибір діапазону дат\",\"todayDate\":\"Сьогодні, {date}\",\"todayDateSelected\":\"Сьогодні, вибрано {date}\"}");


var $c80940728ad8dc0d$exports = {};
$c80940728ad8dc0d$exports = JSON.parse("{\"dateRange\":\"{startDate} 至 {endDate}\",\"dateSelected\":\"已选定 {date}\",\"finishRangeSelectionPrompt\":\"单击以完成选择日期范围\",\"maximumDate\":\"最后一个可用日期\",\"minimumDate\":\"第一个可用日期\",\"next\":\"下一页\",\"previous\":\"上一页\",\"selectedDateDescription\":\"选定的日期：{date}\",\"selectedRangeDescription\":\"选定的范围：{dateRange}\",\"startRangeSelectionPrompt\":\"单击以开始选择日期范围\",\"todayDate\":\"今天，即 {date}\",\"todayDateSelected\":\"已选定今天，即 {date}\"}");


var $ec192a5a83cfafeb$exports = {};
$ec192a5a83cfafeb$exports = JSON.parse("{\"dateRange\":\"{startDate} 至 {endDate}\",\"dateSelected\":\"已選取 {date}\",\"finishRangeSelectionPrompt\":\"按一下以完成選取日期範圍\",\"maximumDate\":\"最後一個可用日期\",\"minimumDate\":\"第一個可用日期\",\"next\":\"下一頁\",\"previous\":\"上一頁\",\"selectedDateDescription\":\"選定的日期：{date}\",\"selectedRangeDescription\":\"選定的範圍：{dateRange}\",\"startRangeSelectionPrompt\":\"按一下以開始選取日期範圍\",\"todayDate\":\"今天，{date}\",\"todayDateSelected\":\"已選取今天，{date}\"}");


$f87cd00fb2ba0f23$exports = {
    "ar-AE": $1fedb39dedbf7c51$exports,
    "bg-BG": $524ada6153e36bf5$exports,
    "cs-CZ": $181053d31fc585a6$exports,
    "da-DK": $87f1f2ee75c25d27$exports,
    "de-DE": $3018d278e45fefcf$exports,
    "el-GR": $880b624ba6c377b6$exports,
    "en-US": $6a7e3934431e9523$exports,
    "es-ES": $4c458303ff44205c$exports,
    "et-EE": $4b6ec4ab19a96fff$exports,
    "fi-FI": $a5184a366ed55e98$exports,
    "fr-FR": $094c11f0389b0f6c$exports,
    "he-IL": $5b28c6b8eb99673e$exports,
    "hr-HR": $519824243c1aad60$exports,
    "hu-HU": $2fde332abe847288$exports,
    "it-IT": $b266f7fa452e8e83$exports,
    "ja-JP": $3528fd03237947e3$exports,
    "ko-KR": $398f3b7902d2708f$exports,
    "lt-LT": $ebaae1c76687311e$exports,
    "lv-LV": $3ecb31d6694563f6$exports,
    "nb-NO": $26dc2e2c103f178c$exports,
    "nl-NL": $6bfd45b83a7d37dd$exports,
    "pl-PL": $6f70d036cac4bd46$exports,
    "pt-BR": $a227ce57c671783d$exports,
    "pt-PT": $97acad4758696d87$exports,
    "ro-RO": $010069c7d7fbbea3$exports,
    "ru-RU": $0ba9c30c48c8d602$exports,
    "sk-SK": $6873e95bd7af7559$exports,
    "sl-SI": $ce3c7852583668e3$exports,
    "sr-SP": $36cbdcce92af5213$exports,
    "sv-SE": $ce53fa032c8abdc9$exports,
    "tr-TR": $358703e281d51beb$exports,
    "uk-UA": $16bef71241a84fd2$exports,
    "zh-CN": $c80940728ad8dc0d$exports,
    "zh-TW": $ec192a5a83cfafeb$exports
};



const $df1d8e967e73ec8e$export$653eddfc964b0f8a = new WeakMap();
function $df1d8e967e73ec8e$export$134cbb7fb09a9522(date) {
    return (date === null || date === void 0 ? void 0 : date.calendar.identifier) === 'gregory' && date.era === 'BC' ? 'short' : undefined;
}
function $df1d8e967e73ec8e$export$b6df97c887c38e1a(state) {
    let formatMessage = $jzHdg$reactariai18n.useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($f87cd00fb2ba0f23$exports)));
    let start, end;
    if ('highlightedRange' in state) ({ start: start , end: end  } = state.highlightedRange || {
    });
    else start = end = state.value;
    let dateFormatter = $jzHdg$reactariai18n.useDateFormatter({
        weekday: 'long',
        month: 'long',
        year: 'numeric',
        day: 'numeric',
        era: $df1d8e967e73ec8e$export$134cbb7fb09a9522(start) || $df1d8e967e73ec8e$export$134cbb7fb09a9522(end),
        timeZone: state.timeZone
    });
    let anchorDate = 'anchorDate' in state ? state.anchorDate : null;
    return $jzHdg$react.useMemo(()=>{
        // No message if currently selecting a range, or there is nothing highlighted.
        if (!anchorDate && start && end) {
            // Use a single date message if the start and end dates are the same day,
            // otherwise include both dates.
            if ($jzHdg$internationalizeddate.isSameDay(start, end)) {
                let date = dateFormatter.format(start.toDate(state.timeZone));
                return formatMessage('selectedDateDescription', {
                    date: date
                });
            } else {
                let dateRange = $df1d8e967e73ec8e$var$formatRange(dateFormatter, formatMessage, start, end, state.timeZone);
                return formatMessage('selectedRangeDescription', {
                    dateRange: dateRange
                });
            }
        }
        return '';
    }, [
        start,
        end,
        anchorDate,
        state.timeZone,
        formatMessage,
        dateFormatter
    ]);
}
function $df1d8e967e73ec8e$export$31afe65d91ef6e8(startDate, endDate, timeZone, isAria) {
    let formatMessage = $jzHdg$reactariai18n.useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($f87cd00fb2ba0f23$exports)));
    let era = $df1d8e967e73ec8e$export$134cbb7fb09a9522(startDate) || $df1d8e967e73ec8e$export$134cbb7fb09a9522(endDate);
    let monthFormatter = $jzHdg$reactariai18n.useDateFormatter({
        month: 'long',
        year: 'numeric',
        era: era,
        calendar: startDate.calendar.identifier,
        timeZone: timeZone
    });
    let dateFormatter = $jzHdg$reactariai18n.useDateFormatter({
        month: 'long',
        year: 'numeric',
        day: 'numeric',
        era: era,
        calendar: startDate.calendar.identifier,
        timeZone: timeZone
    });
    return $jzHdg$react.useMemo(()=>{
        // Special case for month granularity. Format as a single month if only a
        // single month is visible, otherwise format as a range of months.
        if ($jzHdg$internationalizeddate.isSameDay(startDate, $jzHdg$internationalizeddate.startOfMonth(startDate))) {
            if ($jzHdg$internationalizeddate.isSameDay(endDate, $jzHdg$internationalizeddate.endOfMonth(startDate))) return monthFormatter.format(startDate.toDate(timeZone));
            else if ($jzHdg$internationalizeddate.isSameDay(endDate, $jzHdg$internationalizeddate.endOfMonth(endDate))) return isAria ? $df1d8e967e73ec8e$var$formatRange(monthFormatter, formatMessage, startDate, endDate, timeZone) : monthFormatter.formatRange(startDate.toDate(timeZone), endDate.toDate(timeZone));
        }
        return isAria ? $df1d8e967e73ec8e$var$formatRange(dateFormatter, formatMessage, startDate, endDate, timeZone) : dateFormatter.formatRange(startDate.toDate(timeZone), endDate.toDate(timeZone));
    }, [
        startDate,
        endDate,
        monthFormatter,
        dateFormatter,
        formatMessage,
        timeZone,
        isAria
    ]);
}
function $df1d8e967e73ec8e$var$formatRange(dateFormatter, formatMessage, start, end, timeZone) {
    let parts = dateFormatter.formatRangeToParts(start.toDate(timeZone), end.toDate(timeZone));
    // Find the separator between the start and end date. This is determined
    // by finding the last shared literal before the end range.
    let separatorIndex = -1;
    for(let i = 0; i < parts.length; i++){
        let part = parts[i];
        if (part.source === 'shared' && part.type === 'literal') separatorIndex = i;
        else if (part.source === 'endRange') break;
    }
    // Now we can combine the parts into start and end strings.
    let startValue = '';
    let endValue = '';
    for(let i1 = 0; i1 < parts.length; i1++){
        if (i1 < separatorIndex) startValue += parts[i1].value;
        else if (i1 > separatorIndex) endValue += parts[i1].value;
    }
    return formatMessage('dateRange', {
        startDate: startValue,
        endDate: endValue
    });
}





function $02ef492a56b91cb2$export$d652b3ea2d672d5b(props, state) {
    let formatMessage = $jzHdg$reactariai18n.useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($f87cd00fb2ba0f23$exports)));
    let domProps = $jzHdg$reactariautils.filterDOMProps(props);
    let title = $df1d8e967e73ec8e$export$31afe65d91ef6e8(state.visibleRange.start, state.visibleRange.end, state.timeZone, false);
    let visibleRangeDescription = $df1d8e967e73ec8e$export$31afe65d91ef6e8(state.visibleRange.start, state.visibleRange.end, state.timeZone, true);
    // Announce when the visible date range changes
    $jzHdg$reactariautils.useUpdateEffect(()=>{
        // only when pressing the Previous or Next button
        if (!state.isFocused) $jzHdg$reactarialiveannouncer.announce(visibleRangeDescription);
    }, [
        visibleRangeDescription
    ]);
    // Announce when the selected value changes
    let selectedDateDescription = $df1d8e967e73ec8e$export$b6df97c887c38e1a(state);
    $jzHdg$reactariautils.useUpdateEffect(()=>{
        if (selectedDateDescription) $jzHdg$reactarialiveannouncer.announce(selectedDateDescription, 'polite', 4000);
    // handle an update to the caption that describes the currently selected range, to announce the new value
    }, [
        selectedDateDescription
    ]);
    let errorMessageId = $jzHdg$reactariautils.useSlotId([
        Boolean(props.errorMessage),
        props.validationState
    ]);
    // Pass the label to the child grid elements.
    $df1d8e967e73ec8e$export$653eddfc964b0f8a.set(state, {
        ariaLabel: props['aria-label'],
        ariaLabelledBy: props['aria-labelledby'],
        errorMessageId: errorMessageId,
        selectedDateDescription: selectedDateDescription
    });
    // If the next or previous buttons become disabled while they are focused, move focus to the calendar body.
    let nextFocused = $jzHdg$react.useRef(false);
    let nextDisabled = props.isDisabled || state.isNextVisibleRangeInvalid();
    if (nextDisabled && nextFocused.current) {
        nextFocused.current = false;
        state.setFocused(true);
    }
    let previousFocused = $jzHdg$react.useRef(false);
    let previousDisabled = props.isDisabled || state.isPreviousVisibleRangeInvalid();
    if (previousDisabled && previousFocused.current) {
        previousFocused.current = false;
        state.setFocused(true);
    }
    let labelProps = $jzHdg$reactariautils.useLabels({
        id: props['id'],
        'aria-label': [
            props['aria-label'],
            visibleRangeDescription
        ].filter(Boolean).join(', '),
        'aria-labelledby': props['aria-labelledby']
    });
    return {
        calendarProps: $jzHdg$reactariautils.mergeProps(domProps, labelProps, {
            role: 'group',
            'aria-describedby': props['aria-describedby'] || undefined
        }),
        nextButtonProps: {
            onPress: ()=>state.focusNextPage()
            ,
            'aria-label': formatMessage('next'),
            isDisabled: nextDisabled,
            onFocus: ()=>nextFocused.current = true
            ,
            onBlur: ()=>nextFocused.current = false
        },
        prevButtonProps: {
            onPress: ()=>state.focusPreviousPage()
            ,
            'aria-label': formatMessage('previous'),
            isDisabled: previousDisabled,
            onFocus: ()=>previousFocused.current = true
            ,
            onBlur: ()=>previousFocused.current = false
        },
        errorMessageProps: {
            id: errorMessageId
        },
        title: title
    };
}


function $20e845123e697a89$export$3ee915f8151bd4f1(props, state) {
    return $02ef492a56b91cb2$export$d652b3ea2d672d5b(props, state);
}





function $c49ada48cbc48220$export$87e0539f600c24e5(props, state, ref) {
    let res = $02ef492a56b91cb2$export$d652b3ea2d672d5b(props, state);
    // We need to ignore virtual pointer events from VoiceOver due to these bugs.
    // https://bugs.webkit.org/show_bug.cgi?id=222627
    // https://bugs.webkit.org/show_bug.cgi?id=223202
    // usePress also does this and waits for the following click event before firing.
    // We need to match that here otherwise this will fire before the press event in
    // useCalendarCell, causing range selection to not work properly.
    let isVirtualClick = $jzHdg$react.useRef(false);
    let windowRef = $jzHdg$react.useRef(typeof window !== 'undefined' ? window : null);
    $jzHdg$reactariautils.useEvent(windowRef, 'pointerdown', (e)=>{
        isVirtualClick.current = e.width === 0 && e.height === 0;
    });
    // Stop range selection when pressing or releasing a pointer outside the calendar body,
    // except when pressing the next or previous buttons to switch months.
    let endDragging = (e)=>{
        if (isVirtualClick.current) {
            isVirtualClick.current = false;
            return;
        }
        state.setDragging(false);
        if (!state.anchorDate) return;
        let target = e.target;
        let body = document.getElementById(res.calendarProps.id);
        if (body && body.contains(document.activeElement) && (!body.contains(target) || !target.closest('button, [role="button"]'))) state.selectFocusedDate();
    };
    $jzHdg$reactariautils.useEvent(windowRef, 'pointerup', endDragging);
    $jzHdg$reactariautils.useEvent(windowRef, 'pointercancel', endDragging);
    // Also stop range selection on blur, e.g. tabbing away from the calendar.
    res.calendarProps.onBlur = (e)=>{
        if ((!e.relatedTarget || !ref.current.contains(e.relatedTarget)) && state.anchorDate) state.selectFocusedDate();
    };
    // Prevent touch scrolling while dragging
    $jzHdg$reactariautils.useEvent(ref, 'touchmove', (e)=>{
        if (state.isDragging) e.preventDefault();
    }, {
        passive: false,
        capture: true
    });
    return res;
}







function $a07388a797d86b95$export$cb95147730a423f5(props, state) {
    let { startDate: startDate = state.visibleRange.start , endDate: endDate = state.visibleRange.end  } = props;
    let { direction: direction  } = $jzHdg$reactariai18n.useLocale();
    let onKeyDown = (e)=>{
        switch(e.key){
            case 'Enter':
            case ' ':
                e.preventDefault();
                state.selectFocusedDate();
                break;
            case 'PageUp':
                e.preventDefault();
                e.stopPropagation();
                state.focusPreviousSection(e.shiftKey);
                break;
            case 'PageDown':
                e.preventDefault();
                e.stopPropagation();
                state.focusNextSection(e.shiftKey);
                break;
            case 'End':
                e.preventDefault();
                e.stopPropagation();
                state.focusSectionEnd();
                break;
            case 'Home':
                e.preventDefault();
                e.stopPropagation();
                state.focusSectionStart();
                break;
            case 'ArrowLeft':
                e.preventDefault();
                e.stopPropagation();
                if (direction === 'rtl') state.focusNextDay();
                else state.focusPreviousDay();
                break;
            case 'ArrowUp':
                e.preventDefault();
                e.stopPropagation();
                state.focusPreviousRow();
                break;
            case 'ArrowRight':
                e.preventDefault();
                e.stopPropagation();
                if (direction === 'rtl') state.focusPreviousDay();
                else state.focusNextDay();
                break;
            case 'ArrowDown':
                e.preventDefault();
                e.stopPropagation();
                state.focusNextRow();
                break;
            case 'Escape':
                // Cancel the selection.
                if ('setAnchorDate' in state) {
                    e.preventDefault();
                    state.setAnchorDate(null);
                }
                break;
        }
    };
    let visibleRangeDescription = $df1d8e967e73ec8e$export$31afe65d91ef6e8(startDate, endDate, state.timeZone, true);
    let { ariaLabel: ariaLabel , ariaLabelledBy: ariaLabelledBy  } = $df1d8e967e73ec8e$export$653eddfc964b0f8a.get(state);
    let labelProps = $jzHdg$reactariautils.useLabels({
        'aria-label': [
            ariaLabel,
            visibleRangeDescription
        ].filter(Boolean).join(', '),
        'aria-labelledby': ariaLabelledBy
    });
    let dayFormatter = $jzHdg$reactariai18n.useDateFormatter({
        weekday: 'narrow',
        timeZone: state.timeZone
    });
    let { locale: locale  } = $jzHdg$reactariai18n.useLocale();
    let weekDays = $jzHdg$react.useMemo(()=>{
        let weekStart = $jzHdg$internationalizeddate.startOfWeek($jzHdg$internationalizeddate.today(state.timeZone), locale);
        return [
            ...new Array(7).keys()
        ].map((index)=>{
            let date = weekStart.add({
                days: index
            });
            let dateDay = date.toDate(state.timeZone);
            return dayFormatter.format(dateDay);
        });
    }, [
        locale,
        state.timeZone,
        dayFormatter
    ]);
    return {
        gridProps: $jzHdg$reactariautils.mergeProps(labelProps, {
            role: 'grid',
            'aria-readonly': state.isReadOnly || null,
            'aria-disabled': state.isDisabled || null,
            'aria-multiselectable': 'highlightedRange' in state || undefined,
            onKeyDown: onKeyDown,
            onFocus: ()=>state.setFocused(true)
            ,
            onBlur: ()=>state.setFocused(false)
        }),
        headerProps: {
            // Column headers are hidden to screen readers to make navigating with a touch screen reader easier.
            // The day names are already included in the label of each cell, so there's no need to announce them twice.
            'aria-hidden': true
        },
        weekDays: weekDays
    };
}










function $4d833327a32c9193$export$136073280381448e(props, state, ref) {
    let { date: date , isDisabled: isDisabled  } = props;
    let { errorMessageId: errorMessageId , selectedDateDescription: selectedDateDescription  } = $df1d8e967e73ec8e$export$653eddfc964b0f8a.get(state);
    let formatMessage = $jzHdg$reactariai18n.useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($f87cd00fb2ba0f23$exports)));
    let dateFormatter = $jzHdg$reactariai18n.useDateFormatter({
        weekday: 'long',
        day: 'numeric',
        month: 'long',
        year: 'numeric',
        era: $df1d8e967e73ec8e$export$134cbb7fb09a9522(date),
        timeZone: state.timeZone
    });
    let isSelected = state.isSelected(date);
    let isFocused = state.isCellFocused(date);
    isDisabled = isDisabled || state.isCellDisabled(date);
    let isUnavailable = state.isCellUnavailable(date);
    let isSelectable = !isDisabled && !isUnavailable;
    let isInvalid = state.validationState === 'invalid' && ('highlightedRange' in state ? !state.anchorDate && state.highlightedRange && date.compare(state.highlightedRange.start) >= 0 && date.compare(state.highlightedRange.end) <= 0 : state.value && $jzHdg$internationalizeddate.isSameDay(state.value, date));
    if (isInvalid) isSelected = true;
    // For performance, reuse the same date object as before if the new date prop is the same.
    // This allows subsequent useMemo results to be reused.
    let lastDate = $jzHdg$react.useRef(null);
    if (lastDate.current && $jzHdg$internationalizeddate.isEqualDay(date, lastDate.current)) date = lastDate.current;
    lastDate.current = date;
    let nativeDate = $jzHdg$react.useMemo(()=>date.toDate(state.timeZone)
    , [
        date,
        state.timeZone
    ]);
    // aria-label should be localize Day of week, Month, Day and Year without Time.
    let isDateToday = $jzHdg$internationalizeddate.isToday(date, state.timeZone);
    let label1 = $jzHdg$react.useMemo(()=>{
        let label = '';
        // If this is a range calendar, add a description of the full selected range
        // to the first and last selected date.
        if ('highlightedRange' in state && state.value && !state.anchorDate && ($jzHdg$internationalizeddate.isSameDay(date, state.value.start) || $jzHdg$internationalizeddate.isSameDay(date, state.value.end))) label = selectedDateDescription + ', ';
        label += dateFormatter.format(nativeDate);
        if (isDateToday) // If date is today, set appropriate string depending on selected state:
        label = formatMessage(isSelected ? 'todayDateSelected' : 'todayDate', {
            date: label
        });
        else if (isSelected) // If date is selected but not today:
        label = formatMessage('dateSelected', {
            date: label
        });
        if (state.minValue && $jzHdg$internationalizeddate.isSameDay(date, state.minValue)) label += ', ' + formatMessage('minimumDate');
        else if (state.maxValue && $jzHdg$internationalizeddate.isSameDay(date, state.maxValue)) label += ', ' + formatMessage('maximumDate');
        return label;
    }, [
        dateFormatter,
        nativeDate,
        formatMessage,
        isSelected,
        isDateToday,
        date,
        state,
        selectedDateDescription
    ]);
    // When a cell is focused and this is a range calendar, add a prompt to help
    // screenreader users know that they are in a range selection mode.
    let rangeSelectionPrompt = '';
    if ('anchorDate' in state && isFocused && !state.isReadOnly && isSelectable) {
        // If selection has started add "click to finish selecting range"
        if (state.anchorDate) rangeSelectionPrompt = formatMessage('finishRangeSelectionPrompt');
        else rangeSelectionPrompt = formatMessage('startRangeSelectionPrompt');
    }
    let descriptionProps = $jzHdg$reactariautils.useDescription(rangeSelectionPrompt);
    let isAnchorPressed = $jzHdg$react.useRef(false);
    let isRangeBoundaryPressed = $jzHdg$react.useRef(false);
    let touchDragTimerRef = $jzHdg$react.useRef(null);
    let { pressProps: pressProps , isPressed: isPressed  } = $jzHdg$reactariainteractions.usePress({
        // When dragging to select a range, we don't want dragging over the original anchor
        // again to trigger onPressStart. Cancel presses immediately when the pointer exits.
        shouldCancelOnPointerExit: 'anchorDate' in state && !!state.anchorDate,
        preventFocusOnPress: true,
        isDisabled: !isSelectable,
        onPressStart (e) {
            if (state.isReadOnly) {
                state.setFocusedDate(date);
                return;
            }
            if ('highlightedRange' in state && !state.anchorDate && (e.pointerType === 'mouse' || e.pointerType === 'touch')) {
                // Allow dragging the start or end date of a range to modify it
                // rather than starting a new selection.
                // Don't allow dragging when invalid, or weird jumping behavior may occur as date ranges
                // are constrained to available dates. The user will need to select a new range in this case.
                if (state.highlightedRange && !isInvalid) {
                    if ($jzHdg$internationalizeddate.isSameDay(date, state.highlightedRange.start)) {
                        state.setAnchorDate(state.highlightedRange.end);
                        state.setFocusedDate(date);
                        state.setDragging(true);
                        isRangeBoundaryPressed.current = true;
                        return;
                    } else if ($jzHdg$internationalizeddate.isSameDay(date, state.highlightedRange.end)) {
                        state.setAnchorDate(state.highlightedRange.start);
                        state.setFocusedDate(date);
                        state.setDragging(true);
                        isRangeBoundaryPressed.current = true;
                        return;
                    }
                }
                let startDragging = ()=>{
                    state.setDragging(true);
                    touchDragTimerRef.current = null;
                    state.selectDate(date);
                    state.setFocusedDate(date);
                    isAnchorPressed.current = true;
                };
                // Start selection on mouse/touch down so users can drag to select a range.
                // On touch, delay dragging to determine if the user really meant to scroll.
                if (e.pointerType === 'touch') touchDragTimerRef.current = setTimeout(startDragging, 200);
                else startDragging();
            }
        },
        onPressEnd () {
            isRangeBoundaryPressed.current = false;
            isAnchorPressed.current = false;
            clearTimeout(touchDragTimerRef.current);
            touchDragTimerRef.current = null;
        },
        onPress () {
            // For non-range selection, always select on press up.
            if (!('anchorDate' in state) && !state.isReadOnly) {
                state.selectDate(date);
                state.setFocusedDate(date);
            }
        },
        onPressUp (e) {
            if (state.isReadOnly) return;
            // If the user tapped quickly, the date won't be selected yet and the
            // timer will still be in progress. In this case, select the date on touch up.
            // Timer is cleared in onPressEnd.
            if ('anchorDate' in state && touchDragTimerRef.current) {
                state.selectDate(date);
                state.setFocusedDate(date);
            }
            if ('anchorDate' in state) {
                if (isRangeBoundaryPressed.current) // When clicking on the start or end date of an already selected range,
                // start a new selection on press up to also allow dragging the date to
                // change the existing range.
                state.setAnchorDate(date);
                else if (state.anchorDate && !isAnchorPressed.current) {
                    // When releasing a drag or pressing the end date of a range, select it.
                    state.selectDate(date);
                    state.setFocusedDate(date);
                } else if (e.pointerType === 'keyboard' && !state.anchorDate) {
                    // For range selection, auto-advance the focused date by one if using keyboard.
                    // This gives an indication that you're selecting a range rather than a single date.
                    // For mouse, this is unnecessary because users will see the indication on hover. For screen readers,
                    // there will be an announcement to "click to finish selecting range" (above).
                    state.selectDate(date);
                    let nextDay = date.add({
                        days: 1
                    });
                    if (state.isInvalid(nextDay)) nextDay = date.subtract({
                        days: 1
                    });
                    if (!state.isInvalid(nextDay)) state.setFocusedDate(nextDay);
                } else if (e.pointerType === 'virtual') {
                    // For screen readers, just select the date on click.
                    state.selectDate(date);
                    state.setFocusedDate(date);
                }
            }
        }
    });
    let tabIndex = null;
    if (!isDisabled) tabIndex = $jzHdg$internationalizeddate.isSameDay(date, state.focusedDate) ? 0 : -1;
    // Focus the button in the DOM when the state updates.
    $jzHdg$react.useEffect(()=>{
        if (isFocused && ref.current) {
            $jzHdg$reactariautils.focusWithoutScrolling(ref.current);
            // Scroll into view if navigating with a keyboard, otherwise
            // try not to shift the view under the user's mouse/finger.
            // Only scroll the direct scroll parent, not the whole page, so
            // we don't scroll to the bottom when opening date picker popover.
            if ($jzHdg$reactariainteractions.getInteractionModality() !== 'pointer') $jzHdg$reactariautils.scrollIntoView($jzHdg$reactariautils.getScrollParent(ref.current), ref.current);
        }
    }, [
        isFocused,
        ref
    ]);
    let cellDateFormatter = $jzHdg$reactariai18n.useDateFormatter({
        day: 'numeric',
        timeZone: state.timeZone,
        calendar: date.calendar.identifier
    });
    let formattedDate = $jzHdg$react.useMemo(()=>cellDateFormatter.format(nativeDate)
    , [
        cellDateFormatter,
        nativeDate
    ]);
    return {
        cellProps: {
            role: 'gridcell',
            'aria-disabled': !isSelectable || null,
            'aria-selected': isSelected || null,
            'aria-invalid': isInvalid || null
        },
        buttonProps: $jzHdg$reactariautils.mergeProps(pressProps, {
            onFocus () {
                if (!isDisabled) state.setFocusedDate(date);
            },
            tabIndex: tabIndex,
            role: 'button',
            'aria-disabled': !isSelectable || null,
            'aria-label': label1,
            'aria-invalid': isInvalid || null,
            'aria-describedby': [
                isInvalid ? errorMessageId : null,
                descriptionProps['aria-describedby']
            ].filter(Boolean).join(' ') || undefined,
            onPointerEnter (e) {
                // Highlight the date on hover or drag over a date when selecting a range.
                if ('highlightDate' in state && (e.pointerType !== 'touch' || state.isDragging) && isSelectable) state.highlightDate(date);
            },
            onPointerDown (e) {
                // This is necessary on touch devices to allow dragging
                // outside the original pressed element.
                // (JSDOM does not support this)
                if ('releasePointerCapture' in e.target) e.target.releasePointerCapture(e.pointerId);
            },
            onContextMenu (e) {
                // Prevent context menu on long press.
                e.preventDefault();
            }
        }),
        isPressed: isPressed,
        isFocused: isFocused,
        isSelected: isSelected,
        isDisabled: isDisabled,
        isUnavailable: isUnavailable,
        isOutsideVisibleRange: date.compare(state.visibleRange.start) < 0 || date.compare(state.visibleRange.end) > 0,
        isInvalid: isInvalid,
        formattedDate: formattedDate
    };
}
